ACE/ace/config-linux.h

```diff
@@ -204,6 +204,49 @@
 
 #endif /* __UCLIBC__ */
 
+// To support musl (note: musl devs refuse to add a __MUSL__)
+#if !defined (__GLIBC__) && !defined (__UCLIBC__)
+
+// Enable stuff that musl definitly has
+#define ACE_HAS_UCONTEXT_T
+#define ACE_HAS_SIGTIMEDWAIT
+#define ACE_HAS_PTHREADS
+#define ACE_HAS_RECURSIVE_MUTEXES
+#define ACE_HAS_PTHREADS_UNIX98_EXT
+#define ACE_HAS_CPU_SET_T
+#define ACE_HAS_SIGINFO_T
+#define ACE_HAS_SOCKLEN_T
+
+// Mask some features musl lacks
+#define ACE_LACKS_SIGINFO_H
+#define ACE_LACKS_SYS_SYSCTL_H
+#define ACE_LACKS_RWLOCK_T
+#define ACE_LACKS_ISCTYPE
+#define ACE_LACKS_NETDB_REENTRANT_FUNCTIONS
+
+// Following the example set by uclib undef some festures
+#  if defined (ACE_SCANDIR_CMP_USES_VOIDPTR)
+#    undef ACE_SCANDIR_CMP_USES_VOIDPTR
+#  endif /* ACE_SCANDIR_CMP_USES_VOIDPTR */
+
+#  if defined (ACE_SCANDIR_CMP_USES_CONST_VOIDPTR)
+#    undef ACE_SCANDIR_CMP_USES_CONST_VOIDPTR
+#  endif /* ACE_SCANDIR_CMP_USES_CONST_VOIDPTR */
+
+#  if defined(__GLIBC__)
+#    undef __GLIBC__
+#  endif /* __GLIBC__ */
+
+#  if defined(ACE_HAS_SEMUN)
+#    undef ACE_HAS_SEMUN
+#  endif /* ACE_HAS_SEMUN */
+
+// Fixup strange datatypes
+#define __uint32_t uint32_t
+#define loff_t uint32_t
```

Comment:
I think `loff_t` is supposed to have the ability to be 64 bits, so this isn't ideal.

Looking at my copy of musl that comes along with zig, I see this is actually defined by `fcntl.h` if `_GNU_SOURCE` is defined. You would also have to make sure `ace/os_include/sys/os_types.h`, which where `loff_t` is needed, is including `fcntl.h` (maybe if `ACE_HAS_MUSL` is defined). If you define `_GNU_SOURCE` it might also remove the need for some of these other lines as well.

If you don't want to use `_GNU_SOURCE`, then maybe just typedefing `ACE_LOFF_T` as `off_t` for musl might be enough since that's what musl is doing, I think.

Reply:
I'm not particularly fond of defining `_GNU_SOURCE` for all sources that include `ace.h`. I will try your last suggestion and push an update.